NIFI-9248 Use hive-exec:core instead of the "regular" hive-exec dependency

adenes opened the pull request:
- Excluded org.apache.hive:hive-exec and added the "core" classifier version
- Added the removed dependencies explicitly

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

exceptionfactory commented:
Thanks for putting together this update @adenes, it looks like it should reduce the overall size with the removal of unnecessary dependencies, while also avoiding potential runtime conflicts.

For reference, what factors determined which explicit dependencies to include after excluding `hive-exec` from `hive-streaming`?  The Maven configuration for `hive-exec` appears to include a handful of additional dependencies, but are all of those available through currently declared dependencies?  It would be helpful to have some process to track this down in the event of future version updates to `hive-exec`.

adenes commented:
Thanks for the comment, @exceptionfactory 

> For reference, what factors determined which explicit dependencies to include after excluding `hive-exec` from `hive-streaming`? The Maven configuration for `hive-exec` appears to include a handful of additional dependencies, but are all of those available through currently declared dependencies? It would be helpful to have some process to track this down in the event of future version updates to `hive-exec`.

I ran a `mvn dependency:tree` on the `nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors` module and checked which dependencies are missing from the list [1] and added the missing ones as direct dependencies.

I had an ugly oneliner:
`(for a in `cat ~/tmp/hive-shadedlibs`; do ver=$(cat deptree | grep $a | grep -E ":compile|:runtime" | head -1 | grep -Eo ":\d+(\.\d+)*"); echo $a$ver; done) >result`
where 
- `~/tmp/hive-shadedlibs` was manually created from the below linked hive `pom.xml` snippet in the format of `groupId:artifactId` - https://gist.github.com/adenes/7711eb7bd7873002ae46022c84d7cab1
- `deptree` is the output of `mvn dependency:tree -pl nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors`
- the output of this (`result`) should contain all the libraries from the `hive-shadedlibs` with version number - https://gist.github.com/adenes/f5a734be1fb78af9fbf3a49edf05e1fe


[1] https://github.com/apache/hive/blob/rel/release-3.1.2/ql/pom.xml#L925-L961

exceptionfactory commented:
Thanks for the helpful summary @adenes! I have not been able to perform any runtime verification yet, but the approach looks good.

mattyb149 commented:
After the merge conflicts are addressed, you may find that you need the exclusions from NIFI-9283 in additional places

adenes commented:
> After the merge conflicts are addressed, you may find that you need the exclusions from NIFI-9283 in additional places

Thanks @mattyb149 , I have rebased to main and added a new commit with additional log4j exclusions, this should fix the maven enforcer error, let's see what the github actions will show.

mattyb149 commented on nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/pom.xml:
```diff
@@ -97,6 +97,81 @@
+        <dependency>
+            <groupId>org.apache.hive</groupId>
+            <artifactId>hive-storage-api</artifactId>
+            <version>2.7.0</version>
```
Hive 3.1.2 has languished for a while, I'm thinking they will upgrade their dependency as well, but in the meantime we should try it
